be/src/storage/tablet_reader_params.h

```diff
@@ -77,6 +78,8 @@ struct TabletReaderParams {
     bool sorted_by_keys_per_tablet = false;
     OlapRuntimeScanRangePruner runtime_range_pruner;
 
+    std::unordered_map<uint32_t, ColumnAccessPathPtr>* column_access_paths = nullptr;
+
 public:
     std::string to_string() const;
 };
```

Comment:
can we apply late materialization to access paths, for an example, separate the output access paths from the access paths used by predicates.

Reply:
yeah, but the part isn't very clear in my head, so I send path as a expression now, I just want support a framework to read flat map/json

Reply:
Add some TODO for future enhancement.